engine/src/main/java/org/terasology/engine/world/block/shapes/BlockMeshPart.java

```diff
@@ -115,4 +124,103 @@ public BlockMeshPart rotate(Quaternionf rotation) {
 
         return new BlockMeshPart(newVertices, newNormals, texCoords, indices, texFrames);
     }
+
+    private float[] calcLightingValuesForVertexPos(ChunkView chunkView, Vector3f vertexPos, Vector3f normal) {
```

Comment:
whats your reasoning for moving this outside of Chunktessellator?

Reply:
In the FallingBlocks PR, the system generating those meshes first uses the `BlockMeshGenerator`s to generate the mesh data, then edits the data to translate the mesh from world coordinates to local coordinates, then calls `generateOptimisedBuffers`, so if the lighting data were still calculated using the positions available during the `generateOptimisedBuffers` step, it would have been incorrect. This way, it's added during the `BlockMeshGenerator` step, so editing it afterwards works fine.